backend/test_theme_service.py

```
@@ -0,0 +1,365 @@
+                face_height_ratio = 0.32
+                
+                face_width = int(image_width * face_width_ratio)
+                face_height = int(image_height * face_height_ratio)
+                
+                # Ensure minimum face dimensions
+                self.assertGreaterEqual(face_width, 1, f"Face width too small for {image_width}x{image_height}")
+                self.assertGreaterEqual(face_height, 1, f"Face height too small for {image_width}x{image_height}")
+                
+                # Test inner area dimensions
+                inner_width = face_width - int(face_width * 0.10)  # 5% from each side
+                inner_height = face_height - int(face_height * 0.05) + int(face_height * 0.05)  # Top shrink, bottom expand
```

Ruff (0.12.2)

Local variable `inner_height` is assigned to but never used

Remove assignment to unused variable `inner_height`

(F841)